gm4_mountaineering/data/gm4_mountaineering/functions/ski_effects/steep_downhill_ski_speed/check_steep_downhill_ski_speed.mcfunction

```diff
@@ -0,0 +1,13 @@
+# Checks that the player still travels steep downhill (>=7 blocks, or >=5 blocks with poles)
+# @s = player with skiing tag and having fallen >=7 blocks, or >=5 blocks with poles
+# at @s
+# run from main
+
+execute if entity @s[scores={gm4_mountaineering_ski_steep_downhill=1..},predicate=!gm4_mountaineering:holding_poles] unless score @s gm4_mountaineering_vy matches ..-7 run scoreboard players remove @s gm4_mountaineering_ski_steep_downhill 1
+execute if entity @s[scores={gm4_mountaineering_ski_steep_downhill=1..},predicate=gm4_mountaineering:holding_poles] unless score @s gm4_mountaineering_vy matches ..-5 run scoreboard players remove @s gm4_mountaineering_ski_steep_downhill 1
+execute if score @s[tag=gm4_mountaineering_skiing] gm4_mountaineering_vy matches ..-7 run scoreboard players set @s gm4_mountaineering_ski_steep_downhill 4
+execute if score @s[tag=gm4_mountaineering_skiing,predicate=gm4_mountaineering:holding_poles] gm4_mountaineering_vy matches ..-7 run scoreboard players set @s gm4_mountaineering_ski_steep_downhill 4
+
+# Remove speed if no longer travelling downhill or removed skis
+execute at @s[scores={gm4_mountaineering_ski_steep_downhill=..0}] run function gm4_mountaineering:ski_effects/steep_downhill_ski_speed/remove_steep_downhill_ski_speed
+execute at @s[tag=!gm4_mountaineering_skiing] run function gm4_mountaineering:ski_effects/steep_downhill_ski_speed/remove_steep_downhill_ski_speed
```

Comment:
This is a thing in some other files as well, but if you're already executing `at @s` (which this function is according to your header) you only need to use another `at @s` if you teleport the player. Instead, us `if entity @s` here.